docs/developer-docs/build/languages/motoko-index.md

```diff
@@ -0,0 +1,163 @@
+
+ For example, the previous Motoko `Counter` actor has the following Candid interface:
+
+ ``` candid
+ service Counter : {
+   inc : () -> (nat);
+ }
+ ```
+
+ ## Orthogonal persistence
+
+ The Internet Computer persists the memory and other state of your canister as it executes. Thus the state of a Motoko actor, including its in-memory data structures, survive indefinitely. Actor state does not need to be explicitly "restored" and "saved" to external storage, with every message.
+
+ For example, in the following `Registry` actor (canister), that assigns sequential IDs to textual names, the state of the hash table is preserved across calls, even though the state of the actor is replicated across many Internet Computer node machines, and typically not resident in memory.
+
+ ``` motoko
+ import Text "mo:base/Text";
+ import Map "mo:base/HashMap";
+ actor Registry {
+   let map = Map.HashMap<Text, Nat>(10, Text.equal, Text.hash);
+   public func register(name : Text) : async () {
+     switch (map.get(name)) {
+       case null {
+         map.put(name, map.size());
+       };
+       case (?id) { };
+     }
+   };
+   public func lookup(name : Text) : async ?Nat {
+     map.get(name);
+   };
+ };
+ await Registry.register("hello");
+ (await Registry.lookup("hello"), await Registry.lookup("world"))
+ ```
+
+ ## Upgrades
+
+ Motoko provides numerous features to help you leverage orthogonal persistence, including language features that allow you to retain a canister�s data as you upgrade the code of the canister.
+
+ For example, Motoko lets you declare certain variables as `stable`. The values of `stable` variables are automatically preserved across canister upgrades.
+
+ Consider a stable counter:
+
+ ``` motoko
+ actor Counter {
+   stable var value = 0;
+   public func inc() : async Nat {
+     value += 1;
+     return value;
+   };
+ }
+ ```
+
+ It can be installed, incremented *n* times, and then upgraded, without interruption, to, for example, the richer implementation:
+
+ ``` motoko
+ actor Counter {
+   stable var value = 0;
+   public func inc() : async Nat {
+     value += 1;
+     return value;
+   };
+   public func reset() : async () {
+     value := 0;
+   }
+ }
+ ```
+
+ Because `value` was declared `stable`, the current state, *n*, of the service is retained after the upgrade. Counting will continue from *n*, not restart from `0`.
+
+ Because the new interface is compatible with the previous one, existing clients referencing the canister will continue to work, but new clients will be able to exploit its upgraded functionality (the additional `reset` function).
+
+ For scenarios that can�t be solved using stable variables alone, Motoko provides user-definable upgrade hooks that run immediately before and after upgrade, and allow you to migrate arbitrary state to stable variables.
+
+ ## And more �?
```

Comment:
What's with the weird chars?

Reply:
Solved by removing and using doc/md/motoko